### ISSUE TITLE
[Feature] substream or mainstream switch

### ISSUE THREAD

**Issue by spacebares** [elad-bar/ha-blueiris#185]
Any of the following:

- Switch between main or substream in the camera entity.
- Choose during config to use main or substream before cameras are created / updated
- Or just create cameras for both main & substream?

**Comment by Kedryn** [comment]
I think it's for having small live camera entities in HA, when streaming to HA Companion.
For example, i have a small upload bandwidth, and i'm not using live cameras on my HA dashboard because they would fill my internet connection

for smal entities, substream are better.

**Comment by spacebares** [comment]
Yes, for mobile devices which has a good chance to be in not so optimal location with sub par connection, or just want to save on data to take a peek at some camera streams.

In blue iris normally you create a camera that has a sub and a main stream. But this integration only allows viewing of the primary stream url. To get around this you have to create 2 cameras in blue iris, one with primary as main stream and one with primary as sub stream. Completly ignoring the "substream" feature in blue iris. You then have link the two together with "sync with camera" option in order for events to trigger one or the other. This is tedious and very messy, but it works.

Blue iris http(s) stream url allows the use of `&decode=-1` to view the substream of a blue iris camera. What this integration needs is when you go to create a camera and you type in the blueiris config stuff to also have a selector to choose which stream the HomeAssistant camera will use, Main or Sub. It is as simple™ as just using the right blueiris url documented in https://blueirissoftware.com/BlueIris.PDF#page=222 internally somewhere. 

My blue iris trial has long since ran out but i believe **this feature will help out some people who have a dashboard just for viewing cameras with "hidden subtabs" to switch between sub & main stream views like a bootleg video quality selector.**

**Comment by mrdunu** [comment]
+1 keen on this mainstream/substream switch to avoid the CPU overhead of having to duplicate the cameras in BlueIris
